### PR TITLE
[ci] Fix remove sonic-slave docker image issue

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
When build sonic-slave docker, we need remove old docker first.
Current removing code fails when facing some corner cases.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: .azure-pipelines/template-clean-sonic-slave.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+steps:
+- script: |
+    containers=$(docker container ls | grep "sonic-slave" | awk '{ print $1 }')
+    if [ ! -z "$containers" ]; then
+      docker container kill $containers || true
+      sleep 5
+    fi
+    docker images | grep "^<none>" | awk '{print$3}' | xargs -i docker rmi {}
```
Comment (xumia) [inline .azure-pipelines/template-clean-sonic-slave.yml:8]: Can we add a some script to remove the killed containers?
containers=$(docker container ls -a | grep "sonic-slave" | awk '{ print $1 }')
docker container rm containers

Comment (liushilongbuaa) [reply .azure-pipelines/template-clean-sonic-slave.yml:8]: docker container rm 3df0d0091623 dde8eb8bff64
Error response from daemon: You cannot remove a running container 3df0d0091623ecf4237d9ba3b878206af26318e882a15ee8ed9eb0d77b596ae8. Stop the container before attempting removal or force remove

File: slave.mk [hunk 2]
```diff
@@ -875,7 +875,7 @@ $(addprefix $(TARGET_PATH)/, $(DOCKER_IMAGES)) : $(TARGET_PATH)/%.gz : .platform
 		# Save the target deb into DPKG cache
 		$(call SAVE_CACHE,$*.gz,$@)
 	fi
-
+	$(eval $*_CACHE_LOADED:=$*.gz_CACHE_LOADED)
```
Comment (xumia) [inline slave.mk:878]: Need to remove it, right?

File: .azure-pipelines/template-clean-sonic-slave.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+steps:
+- script: |
+    containers=$(docker container ls | grep "sonic-slave" | awk '{ print $1 }')
```
Comment (xumia) [inline .azure-pipelines/template-clean-sonic-slave.yml:3]: docker container ls -a

**Review by xumia — APPROVED** [review]

**Pull request merged**